src/libtools/graphvisitor.cc

```diff
@@ -17,85 +17,90 @@
-  return true;
-}
 
-void GraphVisitor::addEdges_(Graph& g, int vertex){
+bool GraphVisitor::queEmpty() { return true; }
+
+void GraphVisitor::addEdges_(Graph& graph, int vertex) {
   throw runtime_error("addEdges_ method must be defined by your visitor");
 }
 
-void GraphVisitor::exploreNode_(pair<int,GraphNode> &p_gn,Graph& g,Edge ed){
-  explored_.insert(p_gn.first);
+void GraphVisitor::exploreNode(pair<int, GraphNode>& vertex_and_node,
+                                Graph& graph, Edge edge) {
+  explored_.insert(vertex_and_node.first);
 }
 
-vector<int> GraphVisitor::getUnexploredVertex_(Edge ed){
+vector<int> GraphVisitor::getUnexploredVertex(Edge edge) {
   vector<int> unexp_vert;
-  if(explored_.count(ed.getV1())==0){
-    unexp_vert.push_back(ed.getV1());
+  if (explored_.count(edge.getEndPoint1()) == 0) {
+    unexp_vert.push_back(edge.getEndPoint1());
   }
-  if(explored_.count(ed.getV2())==0){
-    unexp_vert.push_back(ed.getV2());
+  if (explored_.count(edge.getEndPoint2()) == 0) {
+    unexp_vert.push_back(edge.getEndPoint2());
   }
   return unexp_vert;
 }
 
-void GraphVisitor::initialize(Graph& g){
-  auto neigh_eds = g.getNeighEdges(startingVertex_);
-  GraphNode gn = g.getNode(startingVertex_);
-  pair<int, GraphNode> p_gn(startingVertex_,gn);
-  exploreNode_(p_gn,g);
-  addEdges_(g, startingVertex_);
+bool GraphVisitor::vertexExplored(int vertex){
+  return explored_.count(vertex)==1;
 }
 
-void GraphVisitor::exec(Graph& g, Edge ed){
-  auto unexp_vert = getUnexploredVertex_(ed);    
+void GraphVisitor::initialize(Graph& graph) {
+  vector<Edge> neigh_eds = graph.getNeighEdges(startingVertex_);
+  GraphNode graph_node = graph.getNode(startingVertex_);
+  pair<int, GraphNode> vertex_and_graph_node(startingVertex_, graph_node);
+  exploreNode(vertex_and_graph_node, graph);
+  addEdges_(graph, startingVertex_);
+}
+
+void GraphVisitor::exec(Graph& graph, Edge edge) {
+  vector<int> unexp_vert = getUnexploredVertex(edge);
   // If no vertices are return than just ignore it means the same
   // vertex was explored from a different direction
-  if(!unexp_vert.size()) return;
-  // If two values are returned this is a problem 
-  if(unexp_vert.size()>1){
-    throw runtime_error("More than one unexplored vertex in an edge,"
-      " did you set the starting node");
+  if (!unexp_vert.size()) return;
+  // If two values are returned this is a problem
+  if (unexp_vert.size() > 1) {
+    throw runtime_error(
+        "More than one unexplored vertex in an edge,"
+        " did you set the starting node");
   }
-  pair<int,GraphNode> pr(unexp_vert.at(0),g.getNode(unexp_vert.at(0)));
- 
-  exploreNode_(pr,g,ed);
 
+  pair<int, GraphNode> vertex_and_node(unexp_vert.at(0),
+                                       graph.getNode(unexp_vert.at(0)));
+
+  exploreNode(vertex_and_node, graph, edge);
 }
 
-Edge GraphVisitor::getEdge_(Graph g){
+Edge GraphVisitor::getEdge_(Graph graph) {
   throw runtime_error("The getEdge_ function must be set");
```

Comment:
?? why have the function ?

Reply:
How do you do that? Leave the function empty?

Reply:
virtual Edge GraphVisitor::getEdge(Graph g)=0;

Reply:
Done.